_rules/SC4-1-2-aria-attribute-has-valid-value.md

```diff
@@ -0,0 +1,256 @@
+---
+name: ARIA attribute has valid value
+test_type: atomic
+
+description: |
+   This rule checks that all ARIA 1.1 attributes (`aria-*`) have valid values
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Wilco Fiers
+- Anne Thyme Nørregaard
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) attribute that is placed on an HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies), and where the attribute name starts with `aria-` and where the attribute can be found in [WAI-ARIA 1.1, Definitions of States and Properties](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def).
+
+### Expectation
+
+Each test target has a valid value according to the [value type](https://www.w3.org/TR/wai-aria-1.1/#propcharacteristic_value) for that `aria-*` attribute as specified by [WAI-ARIA 1.1, Definitions of States and Properties](https://www.w3.org/TR/wai-aria-1.1/#state_prop_def).
+
+**Note:** 
+- For value types `ID Reference` and `ID Reference List` this rule does not require that elements with the given id(s) exists in the document. 
+- For value type `URI` ensure that the value matches the [generic URI syntax](https://www.ietf.org/rfc/rfc3986.txt). This rule does not require that the destination URL exists. 
+
+## Assumptions
```

Comment:
These aren't really assumptions. Your note is pretty clear on this part. I think these should be removed.

Reply:
One more thing. There is an assumption here that someone is only using aria- attributes to express logic defined in the ARIA specifications. If someone is using aria- attributes to express custom states / properties that they are using in scripts for other purposes, this rule would be incorrect.

Reply:
Assumption was added, but then removed again after feedback from other side.

Reply:
For reference: https://github.com/auto-wcag/auto-wcag/pull/233#discussion_r221615564